src/instana/instrumentation/pep0249.py

```diff
@@ -2,139 +2,161 @@
+                    if p in self._connect_params[1]
+                ),
+                None,
+            )
             if db_parameter_name:
-                span.set_tag(ext.DATABASE_INSTANCE, self._connect_params[1][db_parameter_name])
-
-            span.set_tag(ext.DATABASE_STATEMENT, sql_sanitizer(sql))
-            span.set_tag(ext.DATABASE_USER, self._connect_params[1]['user'])
-            span.set_tag('host', self._connect_params[1]['host'])
-            span.set_tag('port', self._connect_params[1]['port'])
+                span.set_attribute(
+                    SpanAttributes.DB_NAME,
+                    self._connect_params[1][db_parameter_name],
+                )
+
+            span.set_attribute(SpanAttributes.DB_STATEMENT, sql_sanitizer(sql))
+            span.set_attribute(SpanAttributes.DB_USER, self._connect_params[1]["user"])
+            span.set_attribute("host", self._connect_params[1]["host"])
+            span.set_attribute("port", self._connect_params[1]["port"])
         except Exception as e:
             logger.debug(e)
         return span
 
-    def __enter__(self):
+    def __enter__(self) -> CursorWrapper:
         return self
 
-    def execute(self, sql, params=None):
-        tracer, parent_span, operation_name = get_tracer_tuple()
+    def execute(self, sql, params=None) -> None:
+        tracer, _, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.execute(sql, params)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        with tracer.start_as_current_span(self._module_name) as span:
             try:
-                self._collect_kvs(scope.span, sql)
-
+                self._collect_kvs(span, sql)
                 result = self.__wrapped__.execute(sql, params)
             except Exception as e:
-                if scope.span:
-                    scope.span.log_exception(e)
+                if span:
+                    span.record_exception(e)
                 raise
             else:
                 return result
 
-    def executemany(self, sql, seq_of_parameters):
-        tracer, parent_span, operation_name = get_tracer_tuple()
+    def executemany(self, sql, seq_of_parameters) -> None:
+        tracer, _, operation_name = get_tracer_tuple()
 
         # If not tracing or we're being called from sqlalchemy, just pass through
-        if (tracing_is_off() or (operation_name == "sqlalchemy")):
+        if tracing_is_off() or (operation_name == "sqlalchemy"):
             return self.__wrapped__.executemany(sql, seq_of_parameters)
 
-        with tracer.start_active_span(self._module_name, child_of=parent_span) as scope:
+        with tracer.start_as_current_span(self._module_name) as span:
```

Comment:
Same as previous comment.

Reply:
Same as [this](https://github.com/instana/python-sensor/pull/575#discussion_r1713861703) answer.